.github/workflows/ci.yaml

```diff
@@ -0,0 +1,74 @@
+name: Actions_Pipeline
+
+on:
+  push:
+    branches: [ features/add_implementation ]
+  pull_request:
+    branches: [ features/add_implementation ]
+
+env:
+  IMAGE_NAME: image-angular-react-starter
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  pipeline_of_build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get repository code
+        uses: actions/checkout@v3
+
+      - name: Installing packages Angular
+        run: cd ${{ github.workspace }}/angular && npm install
+      - name: Installing packages React
+        run: cd ${{ github.workspace }}/react && npm install
+
+      - name: Run test
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: cd ${{ github.workspace }}/angular && npm run test
+
+      - name: Build project
+        run: cd ${{ github.workspace }}/angular && npm run build
+
+      - name: Build Docker image
+        run: sudo docker build . -t $IMAGE_NAME
+
+      - name: Login to GitHub container Registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Post image to github packages
+        run: |
+          IMAGE_ID=ghcr.io/krawczowakris/$IMAGE_NAME
+          VERSION=latest

```

Comment:
the version should be always N chars of the SHA1 revision of your commit. I.e. each new build = new image.
that way we can easily rollback if something we deployed had bug. With a fixed single image - we can't do that.